src/subclass/container.rs

```diff
@@ -51,65 +51,59 @@ impl<T: ContainerImpl + ObjectImpl> ContainerImplExt for T {
+            }
         }
     }
 
     fn parent_remove(&self, container: &Container, widget: &Widget) {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .remove
-                .expect("No parent class impl for \"remove\"");
-            f(container.to_glib_none().0, widget.to_glib_none().0)
+            if let Some(f) = (*parent_class).remove {
+                f(container.to_glib_none().0, widget.to_glib_none().0)
+            }
         }
     }
 
     fn parent_check_resize(&self, container: &Container) {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .check_resize
-                .expect("No parent class impl for \"check_resize\"");
-            f(container.to_glib_none().0)
+            if let Some(f) = (*parent_class).check_resize {
+                f(container.to_glib_none().0)
+            }
         }
     }
 
     fn parent_set_focus_child(&self, container: &Container, widget: Option<&Widget>) {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .set_focus_child
-                .expect("No parent class impl for \"set_focus_child\"");
-            f(container.to_glib_none().0, widget.to_glib_none().0)
+            if let Some(f) = (*parent_class).set_focus_child {
+                f(container.to_glib_none().0, widget.to_glib_none().0)
+            }
         }
     }
 
     fn parent_child_type(&self, container: &Container) -> glib::Type {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .child_type
-                .expect("No parent class impl for \"child_type\"");
-            from_glib(f(container.to_glib_none().0))
+            if let Some(f) = (*parent_class).child_type {
+                from_glib(f(container.to_glib_none().0))
+            }
         }
     }
 
     fn parent_get_path_for_child(&self, container: &Container, widget: &Widget) -> WidgetPath {
         unsafe {
             let data = self.get_type_data();
             let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
-            let f = (*parent_class)
-                .get_path_for_child
-                .expect("No parent class impl for \"get_path_for_child\"");
-            from_glib_borrow(f(container.to_glib_none().0, widget.to_glib_none().0))
+            if let Some(f) = (*parent_class).get_path_for_child {
```

Comment:
This doesn't make sense, unless it returns a None existing path, which also doesn't really make sense

Reply:
Alternatively you could return a `gtk_widget_path_new()`, which is the empty path. Should be fine to panic here though, it seems to make no sense to have child elements but no way of returning a path.

Reply:
This should panic indeed, see https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gtk/gtkcontainer.c#L3874-3895. The C version would simply crash.